fe/fe-core/src/main/java/org/apache/doris/nereids/rules/expression/rules/FoldConstantRuleOnBE.java

```diff
@@ -101,7 +101,7 @@ private Expression foldByBE(Expression root, ExpressionRewriteContext context) {
     private Expression replace(Expression root, Map<String, Expression> constMap, Map<String, Expression> resultMap) {
         for (Entry<String, Expression> entry : constMap.entrySet()) {
             if (entry.getValue().equals(root)) {
-                return resultMap.get(entry.getKey());
+                return resultMap.getOrDefault(entry.getKey(), root);
```

Comment:
the problem is why resultMap does not contain entry.getKey().
if we return root, it means fold-const feature failed.